String conversion bench

dhong44 opened the pull request:
Remove old implementation of String Conversion Bench
since the old implementation does not honour
numIterationsPerLoop, it does not work when
numIterationsPerLoop is modified during the benchmark.

Signed-off-by: Daniel Hong <daniel.hong@live.com>

dhong44 committed:
Remove old implementation of StringConversionBench

Remove old implementation of String Conversion Bench
since the old implementation does not honour
numIterationsPerLoop, it does not work when
numIterationsPerLoop is modified during the benchmark.

Signed-off-by: Daniel Hong <daniel.hong@live.com>

dhong44 committed:
Implement String Conversion Bench

Implement a version of StringConversionBench
that honours the numIterationsPerLoop parameter.

Signed-off-by: Daniel Hong <daniel.hong@live.com>

karianna commented on net/adoptopenjdk/bumblebench/string/StringConversionBench.java:
```diff
@@ -34,362 +35,90 @@
-	private static final int MAX_STRING_LENGTH = option("maxStringLen", 877);
-	private static final int MIN_STRING_LENGTH = option("minStringLen", 877);
+	private static final int STRING_LENGTH = option("stringLength", 0);
+	private static final boolean IS_TESTING_TO_UPPER = option("testUpper",false);
```
whitespace after the `,`

karianna commented on net/adoptopenjdk/bumblebench/string/StringConversionBench.java:
```diff
@@ -34,362 +35,90 @@
-				} catch (FileNotFoundException e) {
-					e.printStackTrace();
+		if (STRING_LENGTH <= 0  || STRING_LENGTH > 1000) {
+			for (int i = 0; i < MAX_NUM_STRINGS; i++) {
```
enhanced for loops or collections / streams not possible here?

dhong44 replied:
I don't think enhanced for loops could be used here since i'm assigning new values to each element of `strings`. I think using `Stream.map()` here would be possible but what would the benefit be?

karianna replied:
None :-), was just checking assumptions :-)

dhong44 committed:
Fix whitespace issues in StringConversionBench

Add a space between commas and the next parameter in
calls to option in StringConversionBench.

Signed-off-by: Daniel Hong <daniel.hong@live.com>

dhong44 committed:
Use common StringTestData

Remove declaration of char array and use common data found in
StringTestData class to avoid duplicated code.

Signed-off-by: Daniel Hong <daniel.hong@live.com>

dhong44 commented:
Prereq #5
